stablehlo/testdata/BUILD.bazel

```diff
@@ -46,6 +46,13 @@ expand_template(
             "@llvm-project//llvm:FileCheck",
         ],
         size = "small",
+        tags = ["stablehlo_data_tests"],
```

Comment:
Renamed as suggested. 

Regarding this:
```
bazel build //...
bazel test //...
```
`test_suite` merely bundles them together but shouldn't cause the `bazel test //...` to run them twice. That said, I'd leave the CI script as-is, because it is always possible for new tests to be added but not be included in the test_suite, in which case CI would have a coverage gap if it relies on the test_suite.